### ISSUE TITLE
In mail tab, quickMove reopens mail in new tab after moving - should go to next unread mail instead

### ISSUE THREAD

**Issue by fredstluka** [RealRaven2000/QuickFolders#132]
Shift-M works fine when I'm looking at a list of messages.

But sometimes, I hit Enter while looking at a list of messages, to see a single message in its own Thunderbird tab.

Then, when I hit Shift-M while looking at that message in a Thunderbird tab, it moves the message as expected.  That causes the Thunderbird tab to show the next message.  Perfect!

BUT, then a new Thunderbird tab is opened, showing the message I just moved.  Not what I wanted.  I wanted to just see the next message in the previously existing tab.  No new tab is desired.  I'm done with the old message.  I read it and filed it away.  Don't need to see it again.

Also, it's confusing because the net visible effect is to keep seeing the same message, not noticing that a new tab has been created to house it.  So, it looks at first as though the Shift-M did NOT move the message.

--Fred

Edit: Corrected typo: "next visible effect" --> "net visible effect"
--Fred

**Comment by RealRaven2000** [comment]
This area (single tab viewing and navigation from there) isn't well tested, guess I have to do more of that. In my workflow I only open emails in single tabs if need to store them for later (so they remain on screen when I restart my Computer / next day). I did some improvements in the area when opening mails in their own window, but I think I need to do a lot more testing to make this useful. I would have Shift-M (moving a mail while looking at it in a tab) -  expected to "stay there and do nothing" or "close the tab" - not necessarily do some automatic advancing to the next unread mail, but again I can see that it would be useful.

**Comment by fredstluka** [comment]
Axel,

Yeah, I prefer to open single mails in their own windows instead
of their own tabs.  I did it that way for many years.

But QuickFolders doesn't work at all from a single message window.
Here are my notes on how and why I changed that in Thunderbird
preferences:

         - General Preferences
           - Reading & Display
             - Open messages in = new window (was new tab)
               - [Sat 1/16/2021]
                 Changed back to new tab.  Gives me just as much room to
                 read it as new window, still closes via Cmd-W, and now
                 QuickFolders commands work for filing away messages.
               - [Tue 1/19/2021]
                 One downside is that when I leave and return the tab, it
                 scrolls back to the top.  Doesn't remember my scroll
                 position like a separate window does.


--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> This area (single tab viewing and navigation from there) isn't well 
> tested, guess I have to do more of that.

Yeah, is there a way for you to do automated regression tests?

As a software developer, I couldn't live without them.  But they're
harder to do when testing a GUI than a CLI or an API.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> I would have Shift-M (moving a mail while looking at it in a tab) - 
> expected to "stay there and do nothing" or "close the tab" - not 
> necessarily do some automatic advancing to the next unread mail, but 
> again I can see that it would be useful.

I'd expect it to do the same as if I deleted the message while
viewing it in a tab.  In that case, Thunderbird automatically
shows the next message in the folder, or the next message
in the search list, or in the conversation, or whatever set of
messages I was looking at when I hit Enter to open the
message in its own tab.

Should also respect the behavior of Thunderbird that is
specified by:
- Preferences
   - General
     - Close message window/tab on move or delete

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
> Axel,
> I would have Shift-M (moving a mail while looking at it in a tab) - expected to "stay there and do nothing" or "close the tab" - not necessarily do some automatic advancing to the next unread mail, but again I can see that it would be useful.
> I'd expect it to do the same as if I deleted the message while viewing it in a tab.  

Not me. Because you are not in the context of a list when you open a single message window - you don't see a list. But having said that, you can add the  Previous and Next buttons to the single message toolbar and navigate from there. Anyway for that reason i am treating this as a "new feature" request and not as gap of something that should always have been in QuickFolders. And I don't even consider it as the expected / default behavior even though I can understand that it would be very utilitarian for the way you use Thunderbird. 

So from my POV this is a nice additional feature, that the large majority of users won't expect or use, and possibly a lot of work for me. Still a nice feature. The CEO of postbox (who strictly implements features based on the 80/20 approach - if it doesn't benefit 80% of the users it's not worth wasting money on it) would certainly not implement it because it doesn't make economical sense. If I implement it then I have to be convinced that nothing else on the wishlist is more important / helpful - I need to convert more users to licensees and put a lot of energy into this first, and naturally features that are intuitively easier to find and use will take precedence. 

Still, I would prefer the new functionality too. And there should be the options "go to next message and go to next unread message" - in the later case, what would happen if no unread messages are left? It's a complicated feature when you start thinking about it.

> In that case, Thunderbird automatically shows the next message in the folder, or the next message in the search list, or in the conversation, or whatever set of messages I was looking at when I hit Enter to open the message in its own tab. Should also respect the behavior of Thunderbird that is specified by: - Preferences   - General     - Close message window/tab on move or delete

Didn't find it at first, I believe you mean
- Preferences
  - Reading & Display
    - Close message window/tab on move or delete

ok.

**Comment by RealRaven2000** [comment]
So I was looking at the implementation, turns out it was actually myself who reopened the tab after moving the email (behavior was by design) - see in quickfolders-quickMove.js function `execute()`. 

```
  if (messageDb && !isCopy) {
    // reusing the existing Tab didn't work so we close & re-open msg from the new location.
    // let tab = (QuickFolders.Util.Application=='Thunderbird') ? tabmail.selectedTab : tabmail.currentTabInfo;
    // currentTab.folderDisplay.selectMessage(msgHdr, true);
    setTimeout( 
      function() {
        let msgHdr = messageDb.getMsgHdrForMessageID(theMessage);
        util.logDebugOptional('quickMove', 'reopen mail in tab:' + msgHdr.mime2DecodedSubject );
        QuickFolders.Util.openMessageTabFromHeader(msgHdr);
      }, 1200);
  }
```

I made this optional now and turned it _off_ by default. I also added "go to the next unread mail" as a configurable option (no UI yet)  and turned that one _on_ by default. Test Version for Thunderbird 78:

[QuickFolders-wx-5.5pre10.zip](https://github.com/RealRaven2000/QuickFolders/files/6312472/QuickFolders-wx-5.5pre10.zip)

New configuration settings:

```
extensions.quickfolders.quickMove.reopenMsgTabAfterMove = false
extensions.quickfolders.quickMove.gotoNextUnreadAfterMove = true
```

So this should do the desired behavior of the OP: " I wanted to just see the next message in the previously existing tab."

**Comment by RealRaven2000** [comment]
Added UI to advanced quickMove settings and translated into 17 languages (thanks, google translate):

![image](https://user-images.githubusercontent.com/6314674/114768763-39150700-9d61-11eb-87a4-24467006d418.png)

[QuickFolders-wx-5.5pre15.zip](https://github.com/RealRaven2000/QuickFolders/files/6313285/QuickFolders-wx-5.5pre15.zip)

@fredstluka please test when you get time? For the "old (unwanted) behavior you would have to check boxes the opposite way than in the screenshot.

**Comment by fredstluka** [comment]
Axel,

Wow!  Thanks for the quick responses to all of these bugs
and feature requests.  I've never seen such good customer
support anywhere, in my entire 38-year career.  There are
one or two extraordinary vendors that send me fixes or new
features within a week or so, but never the same or next
day like you've been doing.  Very nice!!!

OK.  I installed QuickFolders-wx-5.5pre15.zip 
<https://github.com/RealRaven2000/QuickFolders/files/6313285/QuickFolders-wx-5.5pre15.zip>.

Looks great!  Thanks!  Without changing either of the new
checkboxes in the settings panel, it does exactly what I
wanted.

Some minor quirks:

a. It moves to the next message, not the next UNREAD message.
     That's fine with me.  Exactly what I wanted, but not what I
     thought you said it would do.

b. The new settings checkbox "Display next unread mail after
      moving current one" seems to have no effect.  It always
      goes to the next message, not the next unread message.
      That's also fine with me, but perhaps not what you wanted.

c. After moving the message, quickMove seems to leave the
     keyboard focus in the search text box.  The box vanishes,
     but typing keystrokes still causes the menu to appear and
     to change appropriately based on the chars you type, even
     though you can't see them.  Also, Shift-M (to move the next
     message which his now appeared in the tab) doesn't work.
     Probably because it's also being directed to the text box.
     Also, Cmd-Z (undo, to move the message back) doesn't work.
     Probably because it's also being directed to the text box.
     Clicking in the message box (to move focus away from the
     text box) makes everything start working properly again.

I did some pretty thorough testing, since I thought I saw odd
behavior once or twice.  But it was all explained by the above,
I think.

Here are the tests I did, all of which came out perfect.  Thanks!

I compared it with the native behavior of TB's Message menu
items:
- Move To
- Move to "..." Again (Opt-Shift-M)

I tried various scenarios involving a TB tab that shows a single
message and compared the behavior of the native TB move vs
QF quickMove.

1. When I'm looking at an unfiltered folder and open a single
     message in a new tab, then move that message to a new
     folder:
     - TB shows the next message
     - QF does the same.  Perfect!

2. When I'm looking at a FILTERED folder (I typed a string in the
     "Filter these messages" text box above the list of messages,
     of hit Cmd-Shift-K, to see only some of the messages in the
     folder), and open a single message in a new tab, then move
     that message to a new folder:
     - TB shows the next filtered message
     - QF does the same.  Perfect!

3. When I'm looking at a conversation view of messages from
     multiple different folders, and open a single message in a new
     tab, then move that message to a new folder:
     - TB shows the next message in the conversation from whatever folder
     - QF does the same.  Perfect!

4. When I'm looking a list of messages from multiple different
     folders generated by "Show results as list" after doing a search,
     and open a single message in a new tab, then move that
     message to a new folder:
     - TB shows the next message in the list from whatever folder
     - QF does the same.  Perfect!

5.  Same as #1, but it's the last message in the folder
     - TB shows the previous message, since there is no next message
     - QF does the same.  Perfect!

6.  Same as #2, but it's the last filtered message in the folder
     - TB shows the previous filtered message, not the message that
        would be next if the filter were ignored
     - QF does the same.  Perfect!

7.  Same as #3, but it's the last message in the conversation
     - TB shows the previous message in the conversation, since
        there is no next message
     - QF does the same.  Perfect!

8.  Same as #4, but it's the last message in the search results list
     - TB shows the previous message in the list, since there is no
        next message
     - QF does the same.  Perfect!

9.  Same as #1, but it's the only message in the folder
     - TB closes the tab since there are no more messages
     - QF does the same.  Perfect!

10.  Same as #2, but it's the only filtered message in the folder
     - TB closes the tab since there are no more filtered messages
     - QF does the same.  Perfect!

11.  Same as #3, but it's the only message in the conversation
     - TB leaves the tab open, but empty since there are no more
        message in the conversation.  Oddly inconsistent with #9
        and #10, but OK.  I won't bother reporting it as a bug to
        Mozilla.
     - QF does the same.  Perfect match with TB!

12.  Same as #4, but it's the only message in the search results list
     - TB leaves the tab open, but empty since there are no more
        message in the list.   Oddly inconsistent with #9 and #10, but
        OK.  I won't bother reporting it as a bug to Mozilla.
     - QF does the same.  Perfect match with TB!

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

I just realized that GitHub does NOT keep me fully informed via
email.  It sends email whenever a comment is posted or a ticket
is closed, but NOT:
- When a ticket title is changed
- When a comment is edited
- When even the original problem description is edited (I think)
- When a ticket is pinned
- Others?

For example, I edited the original description of this ticket to
correct a typo, and added a note at the bottom of the description
saying the following.  I got no email from that.  Did you?

> Edit: Corrected typo: "next visible effect" --> "net visible effect"
> --Fred


Also, you added these 2 additional bits of text to comments that
you'd already added.  I got email for the original comments, but
none for the edits.  So I could easily have never seen edits.  Did
you get email from them?

1:

> Edit: ok. What happens if you don't have it active without 
> QuickFolders disabled? Does it do the correct / expected thing then??

2:

> PS: Not going to create a dedicated UI for the Thunderbird 68 version 
> because I have no "quickMove Advanced Settings" dialog there. If 
> people want to change these advanced settings, they will have to 
> right-click on the quickMove checkbox in the premium features box on 
> the QF Pro tab.


Also, in #1 above, I'm not sure what you mean by "don't have it
active without QuickFolders disabled".  Please explain.

Thanks!
--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
>  I installed QuickFolders-wx-5.5pre15.zip <https://github.com/RealRaven2000/QuickFolders/files/6313285/QuickFolders-wx-5.5pre15.zip>. Looks great!  Thanks!  Without changing either of the new checkboxes in the settings panel, it does exactly what I wanted. 

> Some minor quirks: a. It moves to the next message, not the next UNREAD message.     That's fine with me.  

not what I wanted. I have to fix that or change all my translations. But I think going to the unread message makes much more sense when you open an isolated special message out of a big list. What are the chances that you want to see exactly the next one?

> Exactly what I wanted, but not what I     thought you said it would do. b. The new settings checkbox "Display next unread mail after      moving current one" seems to have no effect.  It always      goes to the next message, not the next unread message.      That's also fine with me, but perhaps not what you wanted. c. After moving the message, quickMove seems to leave the     keyboard focus in the search text box.  The box vanishes,     but typing keystrokes still causes the menu to appear and     to change appropriately based on the chars you type, even     though you can't see them.  Also, Shift-M (to move the next     message which his now appeared in the tab) doesn't work.     Probably because it's also being directed to the text box.     Also, Cmd-Z (undo, to move the message back) doesn't work.     Probably because it's also being directed to the text box.     Clicking in the message box (to move focus away from the     text box) makes everything start working properly again. I did some pretty thorough testing, since I thought I saw odd behavior once or twice.  But it was all explained by the above, I think. Here are the tests I did, all of which came out perfect.  Thanks! I compared it with the native behavior of TB's Message menu items: - Move To - Move to "..." Again (Opt-Shift-M) I tried various scenarios involving a TB tab that shows a single message and compared the behavior of the native TB move vs QF quickMove. 1. When I'm looking at an unfiltered folder and open a single     message in a new tab, then move that message to a new     folder:     - TB shows the next message     - QF does the same.  Perfect! 2. When I'm looking at a FILTERED folder (I typed a string in the     "Filter these messages" text box above the list of messages,     of hit Cmd-Shift-K, to see only some of the messages in the     folder), and open a single message in a new tab, then move     that message to a new folder:     - TB shows the next filtered message     - QF does the same.  Perfect! 3. When I'm looking at a conversation view of messages from     multiple different folders, and open a single message in a new     tab, then move that message to a new folder:     - TB shows the next message in the conversation from whatever folder     - QF does the same.  Perfect! 4. When I'm looking a list of messages from multiple different     folders generated by "Show results as list" after doing a search,     and open a single message in a new tab, then move that     message to a new folder:     - TB shows the next message in the list from whatever folder     - QF does the same.  Perfect! 5.  Same as (1), but it's the last message in the folder     - TB shows the previous message, since there is no next message     - QF does the same.  Perfect! 6.  Same as (2), but it's the last filtered message in the folder     - TB shows the previous filtered message, not the message that        would be next if the filter were ignored     - QF does the same.  Perfect! 7.  Same as (3), but it's the last message in the conversation     - TB shows the previous message in the conversation, since        there is no next message     - QF does the same.  Perfect! 8.  Same as (4), but it's the last message in the search results list     - TB shows the previous message in the list, since there is no        next message     - QF does the same.  Perfect! 9.  Same as (1), but it's the only message in the folder     - TB closes the tab since there are no more messages     - QF does the same.  Perfect! 10.  Same as (2), but it's the only filtered message in the folder     - TB closes the tab since there are no more filtered messages     - QF does the same.  Perfect! 11.  Same as (3), but it's the only message in the conversation     - TB leaves the tab open, but empty since there are no more        message in the conversation.  Oddly inconsistent with (9)        and (10), but OK.  I won't bother reporting it as a bug to        Mozilla.     - QF does the same.  Perfect match with TB! 12.  Same as (4), but it's the only message in the search results list     - TB leaves the tab open, but empty since there are no more        message in the list.   Oddly inconsistent with (9) and (10), but        OK.  I won't bother reporting it as a bug to Mozilla.     - QF does the same.  Perfect match with TB!

I am having trouble editing the above on github - it's lost all formatting when I quoted it - I guess because it is super long. If I recall correctly you also noticed in that wall of text above somewhere that the search box still has focus, which is definitely not intended. I already have 4.21 submitted for review and retracted it after I read that, because it's very bad to have an invisible textbox reacting to text input!
So here is a fixed version that addresses both problems (for Tb78) - I am setting focus to the single message straight after moving. I also notice that the keyboard shorcut "N" (next email) simply closes the single message tab - behavior that seems to come straight down from Thunderbird, so I am not going to interfere there - I don't want to reinvent all the wheels on this app.

[Uploading QuickFolders-wx-5.5pre68.zip…]()

**Comment by RealRaven2000** [comment]
> Axel, I just realized that GitHub does NOT keep me fully informed via email.  It sends email whenever a comment is posted or a ticket is closed, but NOT: - When a ticket title is changed - When a comment is edited - When even the original problem description is edited (I think) - When a ticket is pinned - Others? For example, I edited the original description of this ticket to correct a typo, and added a note at the bottom of the description saying the following.  I got no email from that.  Did you?

I don't know, I got so much mail from your tickets that I lost track. My guess is that edits are triggering new mail. I usually only do typo and maybe fix uploads if they get stuck. Not sure when github sends off the first mail but I guess when I press the green "Comment" button. When I get a notification via email I click the link below (well actually I I copy the link because Waterfox isn't great on github) and always read it on the github pages directly. I will see if you have made edits in the meantime.

> Edit: Corrected typo: "next visible effect" --> "net visible effect" --Fred
> Also, you added these 2 additional bits of text to comments that you'd already added.  I got email for the original comments, but none for the edits.  So I could easily have never seen edits.  Did you get email from them? 1:
> Edit: ok. What happens if you don't have it active without QuickFolders disabled? Does it do the correct / expected thing then??

that was badly worded - I think I meant what happens if you disable QuickFolders and then move an email while you have it open in a single message tab? (I guess you have to switch to a tab where the folder is visible for that).

**Comment by RealRaven2000** [comment]
> > Axel, I just realized that GitHub does NOT keep me fully informed via email.  It sends email whenever a comment is posted or a ticket is closed, but NOT: - When a ticket title is changed - When a comment is edited - When even the original problem description is edited (I think) - When a ticket is pinned - Others? For example, I edited the original description of this ticket to correct a typo, and added a note at the bottom of the description saying the following.  I got no email from that.  Did you?
> 
> I don't know, I got so much mail from your tickets that I lost track. My guess is that edits are triggering new mail. I usually only do typo and maybe fix uploads if they get stuck. Not sure when github sends off the first mail but I guess when I press the green "Comment" button. When I get a notification via email I click the link below (well actually I I copy the link because Waterfox isn't great on github) and always read it on the github pages directly. I will see if you have made edits in the meantime.

I meant to write "not triggering". And since I believe that I did not edit that in place this time because it changes the meaning and I know you only read the emails. So, have another email :)

**Comment by RealRaven2000** [comment]
Just testing more. with "go to next unread message" Thunderbird fails to do it if there are no unread messages left in the folder. Getting an uncaught exception in that case, and the single message tab just closes:

![image](https://user-images.githubusercontent.com/6314674/115078709-e7eb4b80-9ef7-11eb-9171-0504a2407ca4.png)

**Comment by fredstluka** [comment]
Axel,

> a. It moves to the next message, not the next UNREAD message.
>
> I think going to the unread message makes much more sense when you 
> open an isolated special message out of a big list. What are the 
> chances that you want to see exactly the next one?

No, that really IS exactly what I want.

Once I've gone to a folder and maybe filtered out some of the
messages, or I've opened a conversation view, or I've done a
search and shown it as a list, what I really want to do next is
to move methodically through that list of messages exactly as
though it was the entire contents of a single folder.

With native TB, in all of those cases, I can see the list of
messages and can see the preview of each message as I move
to it.  And can use:
- f = forward to next message
- b = back to previous message
- n = next unread message
- p = previous unread message

And when I move or delete a message from the list it always
shows me the next message.

And the same with QF.  I can move a message to another folder,
and I see the next message in the list.

Perfect!

But sometimes I don't want to spend any screen space on the
folder tree or the list of messages.  I want the message body to
fill the entire window.  So, instead of just looking at the preview
pane, I hit Enter/Return to see the message in its own tab, with
very little wasted screen space.

In that mode, even though I can't see the list anymore, native TB
still lets me use:
- f = forward to next message
- b = back to previous message
- n = next unread message
- p = previous unread message

And when I move or delete a message from the list it always
shows me the next message.  Exactly the same behavior as
when I was using preview pane.

Perfect!

And so with QF, I want the same behavior.  When I move a
message to another folder, I want to see the next message in
the list.

I don't want to have to think about whether I moved it via QF
or via native TB.  And I don't want to have to think about
whether I'm in preview mode or single message mode.  I want
it to behave the same in all of those cases.  When I move a
message, show me the next message.  If I want to jump to the
next unread, I can hit "n" as I do in all the other cases.

Make sense?


> b. The new settings checkbox "Display next unread mail after
>     moving current one" seems to have no effect.  It always goes
>     to the next message, not the next unread message.  That's
>     also fine with me, but perhaps not what you wanted.

I'm not sure you saw this part of my message.  It's probably
the source of the problem.  I assume you were intending to
support 2 modes:
- Display next mail after moving current one
- Display next UNREAD mail after moving current one

Right?  Looks like the code to manage the option is wrong,
so it's always doing the first mode (next mail).


> c. After moving the message, quickMove seems to leave the
>     keyboard focus in the search text box.  The box vanishes,
>     but typing keystrokes still causes the menu to appear and
>     to change appropriately based on the chars you type, even
>     though you can't see them.  Also, Shift-M (to move the next
>     message which his now appeared in the tab) doesn't work.
>     Probably because it's also being directed to the text box.
>     Also, Cmd-Z (undo, to move the message back) doesn't work.
>     Probably because it's also being directed to the text box.
>     Clicking in the message box (to move focus away from the
>     text box) makes everything start working properly again. 

I hope the above is more readable this time.  You did seem to
see at least part of the text, enough to realize that you should
retracted 4.21 until you're sure it's fixed.


> I did some pretty thorough testing, since I thought I saw odd
> behavior once or twice.  But it was all explained by the above,
> I think. Here are the tests I did, all of which came out perfect.
> Thanks!  ...

If you're interested in all the details of the 12 scenarios I
tested, you'll have to read it on GitHub.  I'm not sure if GitHub's
mailer will trash it as it sends, like the GitHub quoting
mechanism seems to do.  I don't think the problem is the length
of the message.  GitHub may be having trouble with the hard
linebreaks I embedded in the text.  It seems to be using
Markdown or something, but not quite getting it right.

Anyhow, on GitHub my comment is well formatted, so the
handling of incoming email text seems OK.  I think the handling
of outgoing email is OK too.  It's just the quoting of one reply
for use in another that seems to munge the text so badly.
Sorry for the inconvenience...  I've seen this kind of problem
with other tools, and it's usually caused by the hard linebreaks
and my use of hyphens as list bullets.


> So here is a fixed version that addresses both problems (for Tb78) - I 
> am setting focus to the single message straight after moving.
>

OK.  I'll try it out.


> I also notice that the keyboard shorcut "N" (next email) simply closes 
> the single message tab - behavior that seems to come straight down 
> from Thunderbird, so I am not going to interfere there - I don't want 
> to reinvent all the wheels on this app.
>

No.  For me, it moves to the next unread in the list, as I
described above.  You're using lowercase "n", not uppercase
"N", right?

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
> > I also notice that the keyboard shorcut "N" (next email) simply closes the single message tab - behavior that seems to come straight down from Thunderbird, so I am not going to interfere there - I don't want to reinvent all the wheels on this app.

> No.  For me, it moves to the next unread in the list, as I described above.  You're using lowercase "n", not uppercase "N", right?

Just the "n" key. no Shift. it happens when there are no unread messages left in the current folder. In folder view, I am used to it jumping to the next unread mail in the next folder (without asking). BUt apparently this throws an exception when you are in single mail view and then just closes the tab.

**Comment by fredstluka** [comment]
Axel,

> Not sure when github sends off the first mail but I guess when I press 
> the green "Comment" button. 

Yeah, I think so.


> I will see if you have made edits in the meantime.

Right.  But what if you read it, then I make edits.  You're not
likely to ever notice them, unless I point them out by explicitly
emailing you (like the case below, where I've now edited one
of my comments after you read it).  When I use Jira, I always
configure it to send me email for edits as well as new
comments, and it does a really nice job of sending me an
email that contains a color-coded diff of the changes.

But Jira really is the best of breed in the issue tracking field.
If you someday decide to move from GitHub to BitBucket,
you'll see what I mean.  It's amazingly better in ways that I
might never have thought of, unless I'd seen it myself.  Just
like TB is amazingly better than MS Outlook for email.  And
just like Mac and Linux are amazingly better than Windows
as an OS.

And just like QF is amazingly better than native TB for
managing folders.  It's always like that with best of breed
apps.  You may not miss them when you've never seen them,
but you'd HATE to give them up once you do.


> what happens if you disable QuickFolders and then move an email while 
> you have it open in a single message tab? (I guess you have to switch 
> to a tab where the folder is visible for that).

No need to disable QuickFolders.  And no need to switch to a
different TB tab.  There are lots ways to file away a message
when viewing it in a tab:
- Menu: Message | Move To
- Menu: Message | Move to "..." Again
- Shortcut: Opt-Cmd-M
- Toolbar: File
- Right-Click: Move To
- Right-Click: Move to "..." Again

They all behave identically.  Also, the native TB Delete (via Mac
Delete or Windows Backspace key) behaves identically with
regard to which message gets shown next.

I just edited the large comment above to say that all of that more
explicitly.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
Here be New versions. had to change all translations and readmes and the way the setting `extensions.quickfolders.quickMove.gotoNextUnreadAfterMove` is stored to `extensions.quickfolders.quickMove.gotoNextMsgAfterMove`. no point in swimming against the stream when it's bound to fail anyway. But I fixed the focus issue  that makes the (invisible) textbox of quickMove / quickJump gobble up all key strokes.

Tb68 version (bumped to 4.21.1 because I already uploaded 4.21): 
[QuickFolders-wx-4.21.1pre27.zip](https://github.com/RealRaven2000/QuickFolders/files/6328038/QuickFolders-wx-4.21.1pre27.zip)
Tb78 version: [QuickFolders-wx-5.5pre71.zip](https://github.com/RealRaven2000/QuickFolders/files/6328039/QuickFolders-wx-5.5pre71.zip)

> No need to disable QuickFolders.  And no need to switch to a
different TB tab.  There are lots ways to file away a message
when viewing it in a tab:
- Menu: Message | Move To
- Menu: Message | Move to "..." Again
- Shortcut: Opt-Cmd-M
- Toolbar: File
- Right-Click: Move To
- Right-Click: Move to "..." Again

no I meant from a SINGLE MESSAGE TAB. that's the whole point. you will get interesting results. But anyway I have given up on going to the next unread message because it's volatile and I am not going to reinvent wheels for Tb

**Comment by fredstluka** [comment]
Axel,

> I meant to write "not triggering". And since I believe that I did not 
> edit that in place this time because it changes the meaning and I know 
> you only read the emails. So, have another email :)

Yeah, I got another email.  Thanks!  I'm glad you corrected that.

Like I said, I'm pretty sure that edits of the title, description,
comments, etc., do NOT trigger emails.  Nor do most other
changes:
- Pinning
- Assigning
- Mentioning
- Adding labels
- Attaching files
- Changing priority
- Changing severity
- Changing value of a custom field you defined for the project
- Changing status from Submitted to Assigned, to Resolved or
    Deferred, to Closed or Re-opened (or any custom status you
    may define for the project)
- Specifying/changing which release it was or will be resolved in
- etc.

In fact, I'm not sure some of those events and possibilities even
exist in GitHub.

With Jira, you can pick and choose which events trigger emails.
And to which groups of people (creator, assignee, watchers,
admins, arbitrary other addresses, etc.)  I always configure it
to email for every change, and to send to other classes of users
or individual users as they prefer.

With GitHub, I suspect the only emails are on Create, Add
comment, and Close.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> no I meant from a SINGLE MESSAGE TAB. that's the whole point. 

I think you didn't understand what I meant to say.  All of these
ARE possible in native TB in a SINGLE MESSAGE TAB:

  * Menu: Message | Move To
  * Menu: Message | Move to "..." Again
  * Shortcut: Opt-Cmd-M
  * Toolbar: File
  * Right-Click: Move To
  * Right-Click: Move to "..." Again


Once I've opened a message in a single message tab, I can't see
the folder tree anymore, but I CAN still use keyboard shortcuts,
and mouse right clicks.  And (on a Mac anyhow) the TB toolbar is
still visible in the tab, so I can use the File button that I dragged
to the TB toolbar from its list of customizable buttons.  And (on
a Mac) the menu is still visible, so I can still use it.

The only thing NOT available in a single message tab is the
folder tree.  Is it different on Windows?  I haven't used TB on
Windows for about 10 years, so I don't remember anymore.


> no point in swimming against the stream when it's bound to fail anyway.
>
> ...
>
> you will get interesting results.
>

Uh, oh!  That doesn't sound good...   Is there a problem lurking
here that I didn't think of?


> But anyway I have given up on going to the next unread message because 
> it's volatile and I am not going to reinvent wheels for Tb
>

OK.

BTW, I hope I'm not coming across as too critical or too pushy.
I'm not trying to tell you how to write your add-on for TB.  QF
is GREAT.  Like I said in another recent email, it's "best of breed".
There's nothing else anywhere near as good as it for managing
TB folders.  I'm THRILLED to be able to use it.

I'm just trying to point out patterns of usage that I, and perhaps
other of your users, follow and how QF can be even better for us.

Some of your recent emails sound a little annoyed, so I hope I
haven't gone too far.  I tend to be a very detail-oriented anal-
retentive kind of guy, and sometimes I don't know when to leave
well enough alone.  But QF is so nearly PERFECT, and you seem
amenable to the tweaks I've suggested, so I keep on polishing.

I'm sorry if I've come across as rude.  Keep up the good work!

I'll try out the latest version soon.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> Edit: scratch that last bit - I am fried after debugging for 2 hours

I saw this at GitHub.  Didn't get it via email.  I can sympathize.
I've been known to get so caught up in coding and especially
debugging that I stay at work all night.  For decades, my office
mates have known that if they show up at 8am and I'm in the
office, I'm not ALREADY there.  I'm STILL there from the night
before.

Today was like that too.  I got so interested in the QF bugs
and features that you were dealing with so fast that I totally
forgot to eat any food.  It's now 7:10 pm.  I've been up for
over 12 hours and I'm just now about to eat breakfast!

Some days are like that...

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

I tried out QuickFolders-wx-5.5pre71.zip 
<https://github.com/RealRaven2000/QuickFolders/files/6328039/QuickFolders-wx-5.5pre71.zip>

Looks good, but 2 minor nits:

- The option "Display next mail after moving current one"
    seems to not work properly.  Whether it's checked or not,
    QF always moves to the next mail.  Never to the next
    unread email, and never just closes the tab.  What is
    expected to happen when this checkbox is off?

- When "Display next mail after moving current one" is
    checked, the keyboard focus problem is fixed.  But when
    it's NOT checked, the focus still stays in the invisible text
    box

Done for the night.  More tomorrow...

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> Just testing more. with "go to next unread message" Thunderbird fails 
> to do it if there are no unread messages left in the folder. Getting 
> an uncaught exception in that case, and the single message tab just 
> closes:

I don't even think I saw this happen, but I wasn't watching the
error console, so I might have missed it.  Is this something I
can even try to reproduce with the latest version?  I'd be glad
to do some more testing if you like.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,
    
    Just the "n" key. no Shift. it happens when
      there are no unread messages left in the current folder
      ...
       BUt apparently this throws an exception when you are in single
      mail view and then just closes the tab.
    
    
    Even with all of my 12 test scenarios, I don't think I ever tested
    this case.  I tested single message tab view that I arrived at from
    4 different contexts:
    - Normal folder of mail messages
    - Filtered folder of mail messages
    - Conversation view of messages from multiple different folders
    - Search list of messages from multiple different folders
    
    And I tested each of them for 3 cases:
    - There is a next message
    - There is not a next, but there is a previous
    - There are no other messages (it's the only one)
    
    I didn't ever test:
    - There is a next message, but not a next UNREAD message
    
    That's because I was focused on next message, not next unread
    message.
    
    I can test it if you like.  Which version should I use, and which 
    options should I have set?
    
    
    In folder view, I am used to it jumping to
      the next unread mail in the next folder (without asking).
    
    Jumping to the next folder is something I've always been prompted
    for.  I saw recently that QF allows me to set an option to change
    this behavior of TB.  I haven't moved to that mode yet.  I've gotten
    used to the extra prompt before going on to the next folder.  But,
    yeah, I might prefer to not have it.  We'll see.  Thanks for
    providing
    the option!
    
    BTW, I think of QF as offering me additional UI actions, like
    dragging
    a message to the QF toolbar, using the new shortcut keys, etc.  It 
    hadn't occurred to me to use QF options to change the behavior of
    standard TB UI actions.  That's a powerful new aspect that might be
    very useful in some cases.  I'll have to keep it in mind.  Where
    else
    does QF do that?
    
    OK.  I'm all caught up on my QF emails finally.  Sorry for taking so
    long to get back to you on some of them.  Couldn't sleep, so I got
    up at 2am.  May go back to sleep for w few hours soon...
    
    --Fred
      
      Fred Stluka -- http://bristle.com -- Glad to be of service!
      
      Open Source: Without walls and fences, we need no Windows or
      Gates.
      
      
    
    On 4/16/21 5:39 PM, RealRaven2000
      wrote:
    
    
      
      
        
          I also notice that the keyboard shorcut "N" (next email)
            simply closes the single message tab - behavior that seems
            to come straight down from Thunderbird, so I am not going to
            interfere there - I don't want to reinvent all the wheels on
            this app.
        
      
      
        No.  For me, it moves to the next unread in the list, as I
          described above.  You're using lowercase "n", not uppercase
          "N", right?
      
      Just the "n" key. no Shift. it happens when there are no unread
        messages left in the current folder. In folder view, I am used
        to it jumping to the next unread mail in the next folder
        (without asking). BUt apparently this throws an exception when
        you are in single mail view and then just closes the tab.
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/RealRaven2000/QuickFolders/issues/132#issuecomment-821577455",
"url": "https://github.com/RealRaven2000/QuickFolders/issues/132#issuecomment-821577455",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by fredstluka** [comment]
Test message, please ignore.
    
    Just testing to see if I found a bug in TB.  I got tired of the
    prompt
    about HTML or plain text format each time I replied to a GitHub
    email.  So I configured TB to always send HTML format to GitHub.
    But the message that I got from GitHub as a result does not show
    quoted text the way it used to.  Bug?  An experiment below...
    
    This is pasted as a quotation via
      Cmd-Shift-O.  Should be
      shown at GitHub with a bar next to it.  And in the email that
      GitHub sends with a bar next to it.
    
    
    This is unquoted text.  No bar shown.
    
    This should be the last line (unless you hit the "..."), since the 
    rest is my .sig file and the message I'm relying to.  Did it work?
    
    --Fred
       Fred Stluka -- http://bristle.com
      -- Glad to be of service! 
      Open Source: Without walls and fences, we need no Windows or
      Gates.
       
    
    On 4/16/21 5:39 PM, RealRaven2000
      wrote:
    
    
      
      
        
          I also notice that the keyboard shorcut "N" (next email)
            simply closes the single message tab - behavior that seems
            to come straight down from Thunderbird, so I am not going to
            interfere there - I don't want to reinvent all the wheels on
            this app.
        
      
      
        No.  For me, it moves to the next unread in the list, as I
          described above.  You're using lowercase "n", not uppercase
          "N", right?
      
      Just the "n" key. no Shift. it happens when there are no unread
        messages left in the current folder. In folder view, I am used
        to it jumping to the next unread mail in the next folder
        (without asking). BUt apparently this throws an exception when
        you are in single mail view and then just closes the tab.
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/RealRaven2000/QuickFolders/issues/132#issuecomment-821577455",
"url": "https://github.com/RealRaven2000/QuickFolders/issues/132#issuecomment-821577455",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by fredstluka** [comment]
Test message 2, please ignore.

Just testing to see if I found a bug in TB.  I got tired of the prompt
about HTML or plain text format each time I replied to a GitHub
email.  So I configured TB to always send HTML format to GitHub.
But the message that I got from GitHub as a result does not show
quoted text the way it used to.  Bug?  An experiment below...

> This is pasted as a quotation via Cmd-Shift-O.  Should be
> shown at GitHub with a bar next to it.  And in the email that
> GitHub sends with a bar next to it.

This is unquoted text.  No bar shown.

This should be the last line (unless you hit the "..."), since the
rest is my .sig file and the message I'm relying to.  Did it work?

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> Just the "n" key. no Shift. it happens when there are no unread 
> messages left in the current folder
> ...
>  BUt apparently this throws an exception when you are in single mail 
> view and then just closes the tab.

Even with all of my 12 test scenarios, I don't think I ever tested
this case.  I tested single message tab view that I arrived at from
4 different contexts:
- Normal folder of mail messages
- Filtered folder of mail messages
- Conversation view of messages from multiple different folders
- Search list of messages from multiple different folders

And I tested each of them for 3 cases:
- There is a next message
- There is not a next, but there is a previous
- There are no other messages (it's the only one)

I didn't ever test:
- There is a next message, but not a next UNREAD message

That's because I was focused on next message, not next unread
message.

I can test it if you like.  Which version should I use, and which
options should I have set?


> In folder view, I am used to it jumping to the next unread mail in the 
> next folder (without asking).

Jumping to the next folder is something I've always been prompted
for.  I saw recently that QF allows me to set an option to change
this behavior of TB.  I haven't moved to that mode yet.  I've gotten
used to the extra prompt before going on to the next folder.  But,
yeah, I might prefer to not have it.  We'll see.  Thanks for providing
the option!

BTW, I think of QF as offering me additional UI actions, like dragging
a message to the QF toolbar, using the new shortcut keys, etc.  It
hadn't occurred to me to use QF options to change the behavior of
standard TB UI actions.  That's a powerful new aspect that might be
very useful in some cases.  I'll have to keep it in mind.  Where else
does QF do that?

OK.  I'm all caught up on my QF emails finally.  Sorry for taking so
long to get back to you on some of them.  Couldn't sleep, so I got
up at 2am.  May go back to sleep for w few hours soon...

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Test message 3.  Please ignore.

Looks like it's a GitHub bug, not a TB bug.

When incoming mail is sent as:
   Content-Type: multipart/alternative;
GitHub correctly processes the section for:
   Content-Type: text/html; charset=utf-8

But when it's sent as simply:
   Content-Type: text/html; charset=utf-8
GitHub seems to strip out all of the HTML tags, and just show
the plain text that was embedded among the HTML.  Weird!

I confirmed this by sending the exact same message to GitHub
(except for changing "Test message" to "Test message 2") in
both formats, and BCC'ing myself.  Then I compared the
raw text of both emails with a diff tool.  One has 2 parts:
text/plain and and text/html.  The other has only the text/html
part.  In both cases, the text/html parts are identical
(except for changing "Test message" to "Test message 2").

But GitHub processes the incoming multi-part message
correctly and the incoming text/html message wrong.

Bummer!  I'll have to keep TB configured to always sent
multi-part to GitHub, which means I have to let it prompt
me on each send.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

Please don't delete or edit the "Test message" or "Test message 2"
comments above.  I found a bug in GitHub's processing of incoming
email and reported it to them at:
- 
https://github.community/t/github-incorrectly-processes-incoming-text-html-email/175071

My bug report includes the links to these 2 comments in this QF repo,
to help the GitHub folks diagnose their problem:
- 
https://github.com/RealRaven2000/QuickFolders/issues/132#issuecomment-821779921
- 
https://github.com/RealRaven2000/QuickFolders/issues/132#issuecomment-821780425

Thanks!
--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
> Looks good, but 2 minor nits: 

thanks for testing! 

> - The option "Display next mail after moving current one"    seems to not work properly.  Whether it's checked or not,    QF always moves to the next mail.  Never to the next    unread email, and never just closes the tab.  What is    expected to happen when this checkbox is off? 

I would expect Thunderbird to close the single message tab - that's the standard behavior when I do nothing and move a viewed mail, and that happened when I didn't force it to re-open the tab (the original behavior) - so the code branch doing that is now dependent on that setting.

https://github.com/RealRaven2000/QuickFolders/blob/ESR78/chrome/content/quickfolders-util.js#L854

`if (quickMove.Settings.isGoNext==true) {... goDoCommand('cmd_nextMsg')};`

that part of the code also contains the setfocus to the single message pane. Unfortunately I also have to do some work when the tab is closed which I didn't think of.

https://github.com/RealRaven2000/QuickFolders/blob/ESR78/chrome/content/quickfolders-quickMove.js#L85

`if (QuickFolders.quickMove.isReopen) { ... QuickFolders.Util.openMessageTabFromHeader(msgHdr); ... }`

> - When "Display next mail after moving current one" is    checked, the keyboard focus problem is fixed.  But when    it's NOT checked, the focus still stays in the invisible text    box Done for the night.  More tomorrow...

I will definitely need to fix that one... I will try to take the focus code out of the condition. Although it would be nicer to have the focus in the message list element if Thunderbird decides to close the tab  - problem here is that the move necessarily happens after the go next command. Need to test this some more. In the meantime I have released 4.21.1 for the Thunderbird 68 users because I am starting a 40% off promo this weekend that lasts until May 1st.

Might have to push another maintenance update (4.21.2) over the weekend if I make more changes. Gotta kick this out the door before uploading a version for Tb78 users, but man do I need to earn money right now...

PS: tried to make this comment by replying to the mail but it didn't arrive. So I am making it here again, which is actually easier now that I am familiar with markdown. I just have to always delete all your signature and the whole quoted stuff at the bottom when I do this which is slightly inconvenient.  Pity that your plain text mails aren't translated to markdown, stuff like - at the start of the line makes nice bullet points.

So if this comment appears duplicated on your side my apologies

**Comment by RealRaven2000** [comment]
Here is a fix for the "Reopen" I think I made a mistake in the line for the isReopen, it should be:

`if (QuickFolders.quickMove.Settings.isReopen) ...`

[QuickFolders-wx-5.5pre73.zip](https://github.com/RealRaven2000/QuickFolders/files/6328921/QuickFolders-wx-5.5pre73.zip)

According to my tests, when the other one (go next) is disabled THunderbird inexplicably goes to the message _after the next_. no idea why. I have seen it close the tab before, so not sure yet why that's going on. It's hard to tell as moving the message is done asynchronously by Tb, so I can't just react directly from my code in this case - I might have to set up an event listener and micro manage whatever Thunderbird does in this case ... which is a recipe for disaster because it forces me to guess so many internal things that may change further down the line. And Shift-M is literally the only case I can control, there are many more ways the message can be moved as you pointed out.

**Comment by RealRaven2000** [comment]
Implemented in in 5.5 / 4.21.2 published today

**Comment by fredstluka** [comment]
> but man do I need to earn money right now...

Hopefully, my recent 2nd donation helps a little.  And
hopefully other users will donate too.  I'll have to start telling
other folks about QF soon.  It's a great product!  I'll do a
mailing soon to promote TB since it's back under the Mozilla
umbrella and I'll promote QF while I'm at it.

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> tried to make this comment by replying to the mail but it didn't arrive.
> ...
> So if this comment appears duplicated on your side my apologies
>

No.  I didn't get it twice.  Only once.  Also, I see it only once
at GitHub.  Your attempt to comment via email must have
failed somehow, I guess.

FYI,
--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

I didn't get around to testing this specific ZIP file, but I AM now
using version 5.5, as of April 19, 2021.

I tested again.  The "Display next mail after moving current one"
now seems to work perfectly.  When ON, it shows the next mail
in the current single-message TB tab.  When OFF, it closes that
tab.  Perfect!

There's now no option, when viewing a message in a
single-message TB tab, and using quickMove, to move to the
next unread message.  But that fine with me.  I didn't need
that.  Can always just hit the native TB "n" key to do so.

The other nit I mentioned is now also fixed.  Regardless of
whether "Display next mail after moving current one" is
checked, the keyboard focus problem no longer occurs.  The
does NOT stay in the invisible text box.  Perfect!

Thanks for the continued excellent support!

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
Axel,

> According to my tests, when the other one (go next) is disabled 
> THunderbird inexplicably goes to the message /after the next/. no idea 
> why. I have seen it close the tab before, so not sure yet why that's 
> going on.

I'm not seeing this problem.  When I uncheck that box, TB reliably
closes the single-message tab, and focus is on the subject line for
the next message.  As expected.

Thanks!
--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------
